source/adapters/opencl/device.cpp

```diff
@@ -907,6 +906,19 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
 
     return UR_RESULT_SUCCESS;
   }
+  case UR_DEVICE_INFO_SUB_GROUP_SIZES_INTEL: {
+    // Have to convert size_t to uint32_t
```

Comment:
Do we need to check that the values fit in 32 bits when potentially downcasting from `size_t`?

Reply:
> Do we need to check that the values fit in 32 bits when potentially downcasting from size_t?

For GPUs, I am not aware of any hardware with sub-group size of >64. We can add an assert if you think it's helpful. But, IMO, if we get a sub-group size of >2^31, a lot of other assumptions will be broken too :)

Reply:
😄